components/checkbox/index.css

```diff
@@ -272,50 +441,34 @@
 	}
 
 	/* Hover for Checked and Indeterminate States */
-	&:hover {
+	&:not(.is-disabled):hover {
 		.spectrum-Checkbox-input:checked + .spectrum-Checkbox-box,
 		&.is-indeterminate .spectrum-Checkbox-input:checked + .spectrum-Checkbox-box,
 		&.is-indeterminate .spectrum-Checkbox-box {
 			&::before {
-				border-color: var(--highcontrast-checkbox-color-hover, var(--mod-checkbox-emphasized-color-hover, var(--spectrum-checkbox-emphasized-color-hover)));
+				border-color: var(--highcontrast-checkbox-highlight-color-hover, var(--mod-checkbox-emphasized-color-hover, var(--spectrum-checkbox-emphasized-color-hover)));
 			}
 		}
 	}
 
-	/* Selected and Indeterminate Focus States */
-	.spectrum-Checkbox-input:focus-visible:checked + .spectrum-Checkbox-box,
-	&.is-indeterminate .spectrum-Checkbox-input:focus-visible + .spectrum-Checkbox-box {
-		&::before {
-			border-color: var(--highcontrast-checkbox-highlight-color-focus, var(--mod-checkbox-emphasized-color-focus, var(--spectrum-checkbox-emphasized-color-focus)));
-		}
-	}
-
-	/* Selected Invalid Focus State */
-	&.is-invalid .spectrum-Checkbox-input:focus-visible:checked + .spectrum-Checkbox-box {
-		&::before {
-			border-color: var(--highcontrast-checkbox-color-default, var(--mod-checkbox-invalid-color-focus, var(--spectrum-checkbox-invalid-color-focus)));
-		}
-	}
-
-	/* TODO: Because this selector was moved to the default variant's styles, this selector block can be deleted when it is safe to make changes to selectors. */
-	/* Invalid Hover States */
-	&.is-invalid:hover {
-		&.is-indeterminate .spectrum-Checkbox-box,
+	&:not(.is-disabled):focus-visible {
+		.spectrum-Checkbox-input:checked + .spectrum-Checkbox-box,
 		&.is-indeterminate .spectrum-Checkbox-input:checked + .spectrum-Checkbox-box,
-		.spectrum-Checkbox-input:checked + .spectrum-Checkbox-box {
+		&.is-indeterminate .spectrum-Checkbox-box {
 			&::before {
-				border-color: var(--highcontrast-checkbox-color-hover, var(--mod-checkbox-invalid-color-hover, var(--spectrum-checkbox-invalid-color-hover)));
+				border-color: var(--highcontrast-checkbox-highlight-color-focus, var(--mod-checkbox-emphasized-color-focus, var(--spectrum-checkbox-emphasized-color-focus)));
 			}
 		}
 	}
 
-	/* Selected and Indeterminate Hover States */
-	&:hover {
+	/* TODO: Because this selector was moved to the default variant's styles, this selector block can be deleted when it is safe to make changes to selectors. */
```

Comment:
This comment was recently added to main and came in with the rebase. Can you see whether we can remove the selector and/or the comment? I'd check against main to make sure about what the comment was referring to is the same.